dev_guide/builds/buildconfig_environment.adoc

```diff
@@ -0,0 +1,147 @@
+[[dev-guide-build-buildconfig-environment]]
+= BuildConfig Environment
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[NOTE]
+====
+You can also manage environment variables defined in the `BuildConfig` with the
+xref:../../dev_guide/environment_variables.adoc#dev-guide-environment-variables[`oc set env`] command.
+====
+
+[[setting-environment-variables]]
+== Setting Environment Variables
+
+[source,yaml]
+----
+env:
+  - name: BUILD_LOGLEVEL <1>
+    value: "5" <2>
+----
+
+<1> The name to assign to the environment variable
+<2> The value to assign to the environment variable
+
+
+[[referencing-other-environment-variables]]
+== Referencing Other Environment Variables
+
+You can reference the value of a previously defined environment
+variable by using the `$()` syntax. If the environment variable
+reference can not be resolved, the value will be set to the provided
+string as shown below.
+
+[source,yaml]
+----
+env:
+  - name: FIRST_ENV
+    value: myvalue
+  - name: SECOND_ENV
+    value: $(FIRST_ENV) <1>
+----
+
+<1> `$(FIRST_ENV)` is expanded to the value of `FIRST_ENV` which is `myvalue` unless `FIRST_ENV` does not exist, in which case the value will be `$(FIRST_ENV)`.
+
+[[escaping-environment-variable-references]]
+== Escaping Environment Variable References
+
+You can escape an environment variable reference by using a double dollar sign.
+[source,yaml]
+----
+env:
+  - name: FIRST_ENV
+    value: myvalue
+  - name: SECOND_ENV
+    value: $$(FIRST_ENV) <1>
+----
+
+<1> `$$(FIRST_ENV)` would be resolved to the string `$(FIRST_ENV)`.
+
+[[using-valuefrom-references]]
+== Using valueFrom References
+
+[[using-value-and-valuefrom-together]]
+=== Using value and valueFrom Together
+
+If both `value` and `valueFrom` are supplied for a build environment variable
+definition, `value` will be used and the `valueFrom` will be ignored.  in the following
+example, the value of `MY_ENV` would be `myvalue` regardless of what the `valueFrom` would
+resolve to.
+
+[source,yaml]
+----
+env:
+  - name: MY_ENV
+    value: myvalue
+    valueFrom:
+      secretKeyRef:
+        name: mysecret
+        key: username
+----
```

Comment:
it seems a bit silly to have this example repeated immediately below.  i'd say this section (which is just an intro to the concept of valuefrom) maybe doesn't need a specific example, given the subsections provide examples.

(but i also think all this content should be refactored to be generic product doc on the behavior of envvars, not build specific, aside from whatever special cases we have in builds)